modules/framework/src-jvm/RunnerCompat.scala

```diff
@@ -140,7 +150,13 @@ trait RunnerCompat[F[_]] { self: sbt.testing.Runner =>
       }
     }
 
-    preventDeadlock(resourceMap(globalResources)).use { read =>
+    val cleanupResource = Resource.make(effect.unit)(_ =>
```

Comment:
I want to bikeshed a little here and use `Resource.onFinalize`, but we can do it later, as long as this PR confirms the fix

Reply:
I've struggled to get the weaver codebase working nicely with intellij.  So I'm finding coding / refactoring a little more difficult that normal.  What IDE are you all using for weaver dev?

Reply:
I exclusively use Metals (with NeoVim, but most of my work on Weaver was done in VS Code with Metals).

Perhaps the issue in IJ is the fact that Weaver has an extremely large build matrix - we have lots of modules that overlap with each other in almost all sources. I remember IJ struggling a lot with that, as it doesn't fit in its model.

The experience in Metals should be nicer, but on the other hand it doesn't have as many refactorings as IJ :) Swings and roundabouts

Reply:
Not sure which exact issue I was tracking, but it's a known problem: https://youtrack.jetbrains.com/issue/SCL-19569

Reply:
In theory, Intellij in BSP should work okay. I think the standard mode makes a lot of assumption that we happen to break in this codebase. Not 100% sure